### ISSUE TITLE
MagicError: name use count (30) exceeded

### ISSUE THREAD

**Issue by reox** [androguard/androguard#439]
Found some APks where some jpeg files produce errors when getting the mime type:
```
magic.MagicException: b'JPEG image data, Exif standard: [TIFF image data, little-endian, direntries=17, height=4032, bps=218, compression=none, PhotometricIntepretation=RGB, manufacturer=NIKON CORPORATION, model=NIKON D3X, orientation=upper-left, width=6048] name use count (30) exceeded
```
see:
```
14:06:09 E:     for f, mime, crc in a.get_files_information():
14:06:09 E:   File "/usr/lib/python3/dist-packages/androguard/core/bytecodes/apk.py", line 486, in get_files_information
14:06:09 E:     yield k, self.get_files_types()[k], self.get_files_crc32()[k]
14:06:09 E:   File "/usr/lib/python3/dist-packages/androguard/core/bytecodes/apk.py", line 452, in get_files_types
14:06:09 E:     self._files[i] = self._get_file_magic_name(buffer)
14:06:09 E:   File "/usr/lib/python3/dist-packages/androguard/core/bytecodes/apk.py", line 423, in _get_file_magic_name
14:06:09 E:     ftype = m.from_buffer(buffer)
14:06:09 E:   File "/usr/lib/python3/dist-packages/magic/__init__.py", line 85, in from_buffer
14:06:09 E:     return self._handle509Bug(e)
14:06:09 E:   File "/usr/lib/python3/dist-packages/magic/__init__.py", line 112, in _handle509Bug
14:06:09 E:     raise e
14:06:09 E:   File "/usr/lib/python3/dist-packages/magic/__init__.py", line 83, in from_buffer
14:06:09 E:     return maybe_decode(magic_buffer(self.cookie, buf))
14:06:09 E:   File "/usr/lib/python3/dist-packages/magic/__init__.py", line 266, in magic_buffer
14:06:09 E:     return _magic_buffer(cookie, buf, len(buf))
14:06:09 E:   File "/usr/lib/python3/dist-packages/magic/__init__.py", line 199, in errorcheck_null
14:06:09 E:     raise MagicException(err)
```

**Comment by reox** [comment]
some bug reports are already in other repos ... seems to be an upstream problem

**Comment by reox** [comment]
btw happening with 
 python3-magic                                         2:0.4.15-1

**Comment by wdormann** [comment]
I ran into this one myself.  So far, it looks like a reasonable workaround is to uninstall python3-magic, and install filemagic instead.  On Ubuntu, it was as simple as "pip3 install filemagic".  On Windows, I had to do that and also get magic.dll in the system path, and also magic.mgc in a path that Python was looking in (in my case, C:\Python36\share\misc).  I got these files from https://github.com/pidydx/libmagicwin64
It may also be possible to extract the windows magic files from https://pypi.org/project/python-magic-bin/0.4.14/, but I haven't tested this.

So yeah, the Windows side was a few more steps than I had hoped, but if python3-magic is broken, using filemagic instead might be a good idea.
